Source/lighting.cpp

```diff
@@ -977,10 +980,11 @@ void ChangeLightOffset(int i, Point position)
 		return;
 	}
 
-	Lights[i]._lunflag = true;
-	Lights[i].position.old = Lights[i].position.tile;
-	Lights[i].oldRadius = Lights[i]._lradius;
-	Lights[i].position.offset = position;
+	auto &lights = Lights[i];
+	lights._lunflag = true;
+	lights.position.old = Lights[i].position.tile;
+	lights.oldRadius = Lights[i]._lradius;

```

Comment:
Two things I noticed is that you generally have called the reference `lights`, but since it's a reference to a single element from the `Lights` array it should be called `light`. I saw you corrected it in once place, but this was the case for all the changes.

You also missed a couple of usages of `Lights[i]` on the right side, like here and on line 971-972.

I corrected this in the rest of the PR.

Other than that it looks good.